### ISSUE TITLE
First Time Coding and Working with Teensy. Need Assistance!

### ISSUE THREAD

**Issue by SlephenX** [playfultechnology/audio-guestbook#122]
Hello! Realizing this is a lot more difficult than I initially realized and need some assistance! What's most difficult for me is not really knowing exactly what I'm supposed to be looking up to figure out where I'm messing up. One thing to note is that I went ahead and did not include a playback button, so no wiring for that. Everything else I believe has been wired per the instructions but I'm fine to revisit.

I believe I got all the wiring done correctly, have tested all connections and worked well. I've made a test .wav file. Downloaded the code, saved the .wav file there (cause I wasn't sure where else I was supposed to put it). So directory is documents>Arduino>audio-guestbook-main>greeting.wav. I went ahead and uploaded the code to the teensy. Had a slight issue with MTP at first and changed USB type to serial + MTP experimental. Had to change the name of "AudioPlaySdWavX" to "AudioPlaySdWav" in order to get it to not have an error. Not sure if that's messing with anything.

Next is that I'm not sure if my headset hook needs to be reversed or not and how to change the code to reflect that.

All in all; where I'm at currently, I can press the button on the teensy, it will redownload everything and then I can hear the initial beep through the headset speaker, which seems to be something. However then nothing else occurs. Having it in the cradle and lifting it does nothing. Putting it back does nothing. It doesn't play the test greeting nor does it make any more beeps. I have no idea if its recording anything because I don't know how to access the SD card at the moment through MTP. I'm getting an SD card reader but don't have that yet.

Thank you in advance and sorry for any ignorance!

**Comment by SlephenX** [comment]
Here are some pictures for context. Note that I do plug in my teensy when testing, its just not plugged in with this picture.

![IMG_0858](https://github.com/playfultechnology/audio-guestbook/assets/73814316/851c633d-8d7f-449a-baff-5b8e0ef5252a)
![IMG_0856](https://github.com/playfultechnology/audio-guestbook/assets/73814316/0a8afe0b-eca3-4268-b52f-e6c5799f7363)
![IMG_0857](https://github.com/playfultechnology/audio-guestbook/assets/73814316/3714f408-7c35-45f2-a6b1-55012ded6a59)

**Comment by SlephenX** [comment]
My issues have been resolved, leaving this comment in case anyone else has issues.

The main issue, of only the initialization beep going and then nothing else coming through was resolved. I looked at the serial monitor and there it showed me that the SD Card could not be read. I went through a rabbit hole looking to see if it was formatting or something else. In the end, I had to change a couple of PIN numbers because I had a revision D audio shield. Correct numbers were 10, 11, and 13 respectively.

Next, with the errors I was getting with MTP, I just had to install it correctly, which I had not done.

Then I was still getting errors with the SD Card and finally I decided to test all my connections from soldering and turns out my ground connection was not good. Redid the connections.

When I finally got everything working after that, the audio was a bit fuzzy, so I redid the connections on the mic and speaker portions and that did it. Sounds crisp and works well.
